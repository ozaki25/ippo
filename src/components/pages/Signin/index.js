import React from 'react';
import styled from 'styled-components';
import propTypes from 'prop-types';
import { Button, Classes, Dialog, Intent } from '@blueprintjs/core';
import ROUTES from 'src/constants/routes';
import Container from 'src/components/templates/Container';
import BasicAuthForm from 'src/components/organisms/BasicAuthForm';

const styles = {
  dialog: {
    margin: '0 15px',
  },
};

const LinkContainer = styled.div`
  font-size: 120%;
  margin: 15px 0;
  text-align: center;
`;

class Signin extends React.Component {
  state = { isOpen: false };

  toggleDialog = () => this.setState(prevState => ({ isOpen: !prevState.isOpen }));

  signin = async ({ data: { email, pass } }) => {
    await this.props.firebase.doSignInWithEmailAndPassword(email, pass);
    this.props.history.push(ROUTES.Menu);
  };

  signup = async ({ data: { email, pass } }) => {
    await this.props.firebase.doCreateUserWithEmailAndPassword(email, pass);
    this.props.history.push(ROUTES.Menu);
  };

  onChange = event => this.setState({ [event.target.name]: event.target.value });

  render() {
    return (
      <Container>
        <BasicAuthForm onSubmit={this.signin} buttonText="ログイン" />
        <LinkContainer>
          <Button onClick={this.toggleDialog} intent={Intent.PRIMARY} minimal large>
            新規登録
          </Button>
        </LinkContainer>
        <Dialog
          title="ユーザ登録"
          isOpen={this.state.isOpen}
          onClose={this.toggleDialog}
          style={styles.dialog}
          isCloseButtonShown
        >
          <div className={Classes.DIALOG_BODY}>
            <BasicAuthForm buttonText="登録" onSubmit={this.signup} />
          </div>
        </Dialog>
      </Container>
    );
  }
}

Signin.displayName = 'Signin';

Signin.propTypes = {
  history: propTypes.shape({
    push: propTypes.func.isRequired,
  }).isRequired,
  firebase: propTypes.shape({
    doSignInWithEmailAndPassword: propTypes.func.isRequired,
  }).isRequired,
};

Signin.defaultProps = {};

export default Signin;
