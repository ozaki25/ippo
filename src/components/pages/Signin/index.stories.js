import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Signin from '.';

const stories = storiesOf('pages/Signin', module);

const props = {
  history: {
    push: action('push'),
    replace: action('replace'),
    goBack: action('goBack'),
  },
  firebase: {
    doSignInWithEmailAndPassword: action('signin', 'signin'),
  },
};

stories.add('通常パターン', () => <Signin {...props} />);
