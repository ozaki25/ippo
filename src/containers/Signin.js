import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { withFirebase } from 'src/context/firebase';
import { compose } from 'recompose';
import { graphql } from 'react-apollo';
import query from 'src/graphql/query';
import Signin from 'src/components/pages/Signin';
import { setAuthUser } from 'src/modules/session';

const mapStateToProps = state => ({
  rawId: state.webauth.rawId,
  uid: state.webauth.uid,
});

const mapDispatchToProps = dispatch => ({
  onSetAuthUser: authUser => dispatch(setAuthUser(authUser)),
});

export default compose(
  withRouter,
  withFirebase,
  connect(mapStateToProps, mapDispatchToProps),
  graphql(query.fetchUser, { name: 'fetchUser' }),
)(Signin);
