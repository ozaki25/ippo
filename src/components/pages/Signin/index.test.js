import React from 'react';
import { snapshot } from 'src/test/helpers';
import Signin from '.';

const props = {
  history: {
    push: jest.fn(),
    replace: jest.fn(),
  },
  firebase: {
    auth: {
      getRedirectResult: jest.fn(),
    },
    doSignInWithEmailAndPassword: jest.fn(),
    doSignInWithGoogle: jest.fn(),
    doSignOut: jest.fn(),
  },
};

// snapshot('Signin/nomal', <Signin {...props} />);
test('', () => expect(true).toBe(true));
