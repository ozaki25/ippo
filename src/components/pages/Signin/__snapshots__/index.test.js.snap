// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Signin/nomal 1`] = `
.c0 {
  padding: 10px 15px;
}

.c1 {
  font-size: 120%;
  margin: 15px 0;
  text-align: center;
}

<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <Signin
    firebase={
      Object {
        "doSignInWithEmailAndPassword": [MockFunction],
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
  >
    <Container
      authUser={null}
    >
      <withRouter()
        authUser={null}
      >
        <Route>
          <Component
            authUser={null}
            history={
              Object {
                "createHref": [Function],
                "path": "",
                "push": [Function],
                "replace": [Function],
              }
            }
            location=""
            match={
              Object {
                "params": Object {},
                "path": "",
                "url": "",
              }
            }
          >
            <NavigationBar
              authUser={null}
              firebase={null}
              history={
                Object {
                  "createHref": [Function],
                  "path": "",
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location=""
              match={
                Object {
                  "params": Object {},
                  "path": "",
                  "url": "",
                }
              }
            >
              <Blueprint3.Navbar>
                <div
                  className="bp3-navbar"
                >
                  <Blueprint3.NavbarGroup
                    align="left"
                  >
                    <div
                      className="bp3-navbar-group bp3-align-left"
                    >
                      <Blueprint3.NavbarHeading>
                        <div
                          className="bp3-navbar-heading"
                        >
                          IPPO
                        </div>
                      </Blueprint3.NavbarHeading>
                      <Blueprint3.NavbarDivider>
                        <div
                          className="bp3-navbar-divider"
                        />
                      </Blueprint3.NavbarDivider>
                    </div>
                  </Blueprint3.NavbarGroup>
                  <Blueprint3.NavbarGroup
                    align="right"
                  >
                    <div
                      className="bp3-navbar-group bp3-align-right"
                    >
                      <Blueprint3.Button
                        icon="wrench"
                        minimal={true}
                        onClick={[Function]}
                      >
                        <button
                          className="bp3-button bp3-minimal"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          type="button"
                        >
                          <Blueprint3.Icon
                            icon="wrench"
                            key="leftIcon"
                          >
                            <span
                              className="bp3-icon bp3-icon-wrench"
                              icon="wrench"
                            >
                              <svg
                                data-icon="wrench"
                                height={16}
                                viewBox="0 0 16 16"
                                width={16}
                              >
                                <desc>
                                  wrench
                                </desc>
                                <path
                                  d="M15.83 3.7l-3.06 3.05-2.84-.7-.7-2.83L12.29.17a5.004 5.004 0 0 0-4.83 1.29 4.967 4.967 0 0 0-1.12 5.36L.58 12.58c-.36.36-.58.86-.58 1.41 0 1.1.9 2 2 2 .55 0 1.05-.22 1.41-.59l5.77-5.77c1.79.69 3.91.33 5.35-1.12 1.32-1.3 1.74-3.15 1.3-4.81z"
                                  fillRule="evenodd"
                                  key="0"
                                />
                              </svg>
                            </span>
                          </Blueprint3.Icon>
                          <Blueprint3.Icon
                            key="rightIcon"
                          />
                        </button>
                      </Blueprint3.Button>
                    </div>
                  </Blueprint3.NavbarGroup>
                </div>
              </Blueprint3.Navbar>
            </NavigationBar>
          </Component>
        </Route>
      </withRouter()>
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "c0",
                "rules": Array [
                  "
  padding: 10px 15px;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0"
          >
            <BasicAuthForm
              buttonText="ログイン"
              onSubmit={[Function]}
            >
              <form>
                <Blueprint3.FormGroup
                  label="メールアドレス"
                  labelFor="email"
                >
                  <div
                    className="bp3-form-group"
                  >
                    <label
                      className="bp3-label"
                      htmlFor="email"
                    >
                      メールアドレス
                       
                      <span
                        className="bp3-text-muted"
                      />
                    </label>
                    <div
                      className="bp3-form-content"
                    >
                      <Blueprint3.InputGroup
                        id="email"
                        large={true}
                        name="email"
                        onChange={[Function]}
                        value=""
                      >
                        <div
                          className="bp3-input-group bp3-large"
                        >
                          <Blueprint3.Icon />
                          <input
                            className="bp3-input"
                            id="email"
                            name="email"
                            onChange={[Function]}
                            style={
                              Object {
                                "paddingRight": 10,
                              }
                            }
                            type="text"
                            value=""
                          />
                        </div>
                      </Blueprint3.InputGroup>
                    </div>
                  </div>
                </Blueprint3.FormGroup>
                <Blueprint3.FormGroup
                  label="パスワード"
                  labelFor="pass"
                >
                  <div
                    className="bp3-form-group"
                  >
                    <label
                      className="bp3-label"
                      htmlFor="pass"
                    >
                      パスワード
                       
                      <span
                        className="bp3-text-muted"
                      />
                    </label>
                    <div
                      className="bp3-form-content"
                    >
                      <Blueprint3.InputGroup
                        id="pass"
                        large={true}
                        name="pass"
                        onChange={[Function]}
                        type="password"
                        value=""
                      >
                        <div
                          className="bp3-input-group bp3-large"
                        >
                          <Blueprint3.Icon />
                          <input
                            className="bp3-input"
                            id="pass"
                            name="pass"
                            onChange={[Function]}
                            style={
                              Object {
                                "paddingRight": 10,
                              }
                            }
                            type="password"
                            value=""
                          />
                        </div>
                      </Blueprint3.InputGroup>
                    </div>
                  </div>
                </Blueprint3.FormGroup>
                <Blueprint3.Button
                  disabled={true}
                  fill={true}
                  large={true}
                  onClick={[Function]}
                  text="ログイン"
                  type="submit"
                >
                  <button
                    className="bp3-button bp3-disabled bp3-fill bp3-large"
                    disabled={true}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    tabIndex={-1}
                    type="submit"
                  >
                    <Blueprint3.Icon
                      key="leftIcon"
                    />
                    <span
                      className="bp3-button-text"
                      key="text"
                    >
                      ログイン
                    </span>
                    <Blueprint3.Icon
                      key="rightIcon"
                    />
                  </button>
                </Blueprint3.Button>
              </form>
            </BasicAuthForm>
            <styled.div>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": true,
                      "lastClassName": "c1",
                      "rules": Array [
                        "
  font-size: 120%;
  margin: 15px 0;
  text-align: center;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="c1"
                >
                  <Blueprint3.Button
                    intent="primary"
                    large={true}
                    minimal={true}
                    onClick={[Function]}
                  >
                    <button
                      className="bp3-button bp3-large bp3-minimal bp3-intent-primary"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      type="button"
                    >
                      <Blueprint3.Icon
                        key="leftIcon"
                      />
                      <span
                        className="bp3-button-text"
                        key="text"
                      >
                        新規登録
                      </span>
                      <Blueprint3.Icon
                        key="rightIcon"
                      />
                    </button>
                  </Blueprint3.Button>
                </div>
              </StyledComponent>
            </styled.div>
            <Blueprint3.Dialog
              canOutsideClickClose={true}
              isCloseButtonShown={true}
              isOpen={false}
              onClose={[Function]}
              style={
                Object {
                  "margin": "0 15px",
                }
              }
              title="ユーザ登録"
            >
              <Blueprint3.Overlay
                autoFocus={true}
                backdropProps={Object {}}
                canEscapeKeyClose={true}
                canOutsideClickClose={true}
                className="bp3-overlay-scroll-container"
                enforceFocus={true}
                hasBackdrop={true}
                isCloseButtonShown={true}
                isOpen={false}
                lazy={true}
                onClose={[Function]}
                style={
                  Object {
                    "margin": "0 15px",
                  }
                }
                title="ユーザ登録"
                transitionDuration={300}
                transitionName="bp3-overlay"
                usePortal={true}
              />
            </Blueprint3.Dialog>
          </div>
        </StyledComponent>
      </styled.div>
    </Container>
  </Signin>
</MockRouter>
`;
